test/CadlRanchProjectsNonAzure.Tests/type-property-value-types.cs

```diff
@@ -0,0 +1,441 @@
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_Model_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetModelClient().PutAsync(new ModelProperty(new InnerModel("hello")));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsString_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsStringClient().GetCollectionsStringAsync();
+            CollectionAssert.AreEqual(new[] { "hello", "world" }, response.Value.Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsString_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsStringClient().PutAsync(new CollectionsStringProperty(new[] { "hello", "world" }));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsInt_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsIntClient().GetCollectionsIntAsync();
+            CollectionAssert.AreEqual(new[] { 1, 2 }, response.Value.Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsInt_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsIntClient().PutAsync(new CollectionsIntProperty(new[] { 1, 2 }));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsModel_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsModelClient().GetCollectionsModelAsync();
+            var result = response.Value;
+            Assert.AreEqual("hello", result.Property[0].Property);
+            Assert.AreEqual("world", result.Property[1].Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsModel_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsModelClient().PutAsync(new CollectionsModelProperty(new[] { new InnerModel("hello"), new InnerModel("world") }));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_DictionaryString_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetDictionaryStringClient().GetDictionaryStringAsync();
+            var result = response.Value;
+            Assert.AreEqual("hello", result.Property["k1"]);
+            Assert.AreEqual("world", result.Property["k2"]);
```

Comment:
we also need to assert the count of `result.Property` is `2`

Reply:
fixed